src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitSchedulerService.kt

```diff
@@ -76,9 +77,33 @@ class VisitSchedulerService(
       available = filterNonAssociationByPrisonerId(available, prisonerId)
     }
 
+    populateBookedCounts(available)
+
     return available.sortedWith(compareBy { it.startTimestamp })
   }
 
+  private fun populateBookedCounts(available: List<VisitSession>) {
+    available.forEach {
+      it.openVisitBookedCount = visitBookedCount(it, VisitRestriction.OPEN)
+      it.closedVisitBookedCount = visitBookedCount(it, VisitRestriction.CLOSED)
+    }
+  }
+
+  fun visitBookedCount(session: VisitSession, restriction: VisitRestriction): Int {
+    return visitRepository.findAll(
+      VisitSpecification(
+        VisitFilter(
+          prisonId = session.prisonId,
+          startDateTime = session.startTimestamp,
+          endDateTime = session.endTimestamp,
+          visitRestriction = restriction
+        )
+      )
+    ).count {
+      it.status == VisitStatus.BOOKED || it.status == VisitStatus.RESERVED
```

Comment:
This is not a fast doing things!

Reply:
This could really hamper live performance... we need to get rid of this!

Reply:
Same backlog item